horovod/run/util/cache.py

```diff
@@ -75,11 +68,26 @@ def put(self, key, val):
         self._lock.acquire()
         self._content[key] = (datetime.datetime.now(), val)
         try:
-            with open(self._cache_file, 'wb') as cf:
-                cloudpickle.dump(self._content, cf)
+            self._dump(self._cache_file)
         finally:
             self._lock.release()
 
+    def _dump(self, content):
+        with open(self._cache_file, 'wb') as cf:
+            cloudpickle.dump(content, cf, protocol=self._pickle_protocol)
+
+    def _load(self):
+        with open(self._cache_file, 'rb') as cf:
+            try:
+                content = cloudpickle.load(cf)
+            except Exception as e:
+                print(
```

Comment:
Should we just delete the file automatically?

Reply:
good idea. Will do.